composer/utils/inference.py

```
@@ -0,0 +1,166 @@
+        with tempfile.TemporaryDirectory() as tempdir:
+            composer_states_filepath, _, _ = download_checkpoint(path=load_path,
+                                                                 node_checkpoint_folder=tempdir,
+                                                                 object_store=load_object_store,
+                                                                 progress_bar=True)
+            state_dict = torch.load(composer_states_filepath, map_location='cpu')
+            missing_keys, unexpected_keys = model.load_state_dict(state_dict['state']['model'], strict=strict)
+            if len(missing_keys) > 0:
+                log.warning(f"Found these missing keys in the checkpoint: {', '.join(missing_keys)}")
+            if len(unexpected_keys) > 0:
+                log.warning(f"Found these unexpected keys in the checkpoint: {', '.join(unexpected_keys)}")
+
+    model.eval()
+    # Apply transformations (i.e., inference optimizations) in the given order
+    for transform in ensure_tuple(transforms):
+        model = transform(model)
+
+    is_remote_store = store_object_store is not None
+    tempdir_ctx = tempfile.TemporaryDirectory() if is_remote_store else contextlib.nullcontext(None)
+    with tempdir_ctx as tempdir:
+        if is_remote_store:
+            local_store_path = os.path.join(str(tempdir), 'model.export')
+        else:
+            local_store_path = store_path
+
+        if target_format == ExportFormat.TORCHSCRIPT:
+            export_model = None
+            try:
+                export_model = torch.jit.script(model)
+            except Exception as e:
+                log.warning(
+                    f'Scripting with torch.jit.script failed with the following exception. Trying torch.jit.trace!'
+                    f' {str(e)}')
+                if sample_input is not None:
+                    export_model = torch.jit.trace(model, sample_input)
+                else:
+                    log.error(
```

in this case can we raise an exception and propagate the original exception? e.g. `raise RuntimeError('Scripting with torch.jit.script failed and sample inputs are not provided for tracing with torch.jit.trace') from e`